### PR TITLE
Run muzzle check against OpenTelemetry API instrumentation (bridge)

### PR DESCRIPTION
**Pull request by trask**
Addresses part of #3044

This PR changes muzzle checks to run against shaded artifacts (similar to how tests are run against shaded artifacts). The main reason for doing this is to enable running muzzle checks against OpenTelemetry API instrumentation.

This will help unblock PR #4786 which is currently failing on `:instrumentation:reactor-3.1:javaagent:muzzle` due to `java.lang.ClassNotFoundException: application.io.opentelemetry.context.ContextStorage`

### CODE REVIEW

File: gradle-plugins/src/main/kotlin/io.opentelemetry.instrumentation.muzzle-check.gradle.kts [hunk 1]
```diff
@@ -38,14 +41,37 @@ val muzzleTooling: Configuration by configurations.creating {
...
   isCanBeResolved = true
 }
+
 val muzzleBootstrap: Configuration by configurations.creating {
   isCanBeConsumed = false
   isCanBeResolved = true
 }
 
+val muzzleShadow by tasks.registering(ShadowJar::class) {
+  from(tasks.jar.get().archiveFile.get())
```
Comment (iNikem) [inline gradle-plugins/src/main/kotlin/io.opentelemetry.instrumentation.muzzle-check.gradle.kts:51]:
```suggestion
  from(tasks.jar)
```

File: gradle-plugins/src/main/kotlin/io.opentelemetry.instrumentation.muzzle-check.gradle.kts [hunk 2]
```diff
@@ -38,14 +41,37 @@ val muzzleTooling: Configuration by configurations.creating {
...
+
+  configurations = listOf(project.configurations.runtimeClasspath.get())
```
Comment (iNikem) [inline gradle-plugins/src/main/kotlin/io.opentelemetry.instrumentation.muzzle-check.gradle.kts:53]: this is default, no?

File: gradle-plugins/src/main/kotlin/io.opentelemetry.instrumentation.muzzle-check.gradle.kts [hunk 3]
```diff
@@ -38,14 +41,37 @@ val muzzleTooling: Configuration by configurations.creating {
   isCanBeConsumed = false
   isCanBeResolved = true
 }
+
 val muzzleBootstrap: Configuration by configurations.creating {
   isCanBeConsumed = false
   isCanBeResolved = true
 }
 
+val muzzleShadow by tasks.registering(ShadowJar::class) {
```
Comment (iNikem) [inline gradle-plugins/src/main/kotlin/io.opentelemetry.instrumentation.muzzle-check.gradle.kts:50]: I don't understand why this shadow jar is needed?

File: gradle-plugins/src/main/kotlin/io.opentelemetry.instrumentation.muzzle-check.gradle.kts (hunk 2 above)
Comment (trask) [reply gradle-plugins/src/main/kotlin/io.opentelemetry.instrumentation.muzzle-check.gradle.kts:53]: defaults are only applied to the default "shadowJar" task, see https://github.com/johnrengelman/shadow/issues/108#issuecomment-62418005

File: gradle-plugins/src/main/kotlin/io.opentelemetry.instrumentation.muzzle-check.gradle.kts (hunk 3 above)
Comment (trask) [reply gradle-plugins/src/main/kotlin/io.opentelemetry.instrumentation.muzzle-check.gradle.kts:50]: this shades the javaagent module classes themselves

Comment (iNikem) [reply gradle-plugins/src/main/kotlin/io.opentelemetry.instrumentation.muzzle-check.gradle.kts:50]: Rename it to `shadedModule` then? Or something similar. We don't shade muzzle here, do we?

**Review by mateuszrzeszutek — APPROVED** [review]

**Review by iNikem — APPROVED** [review]

**Pull request merged**